fim_process_unit_wb.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash -e
+
+# Why is this file here and it appears to be using duplicate export variables?
+# For AWS, we need to make a direct call to this files with two params, hucNumber first, 
+# then the runName same as the -n flag in fim_pipeline and fim_pre_processing.sh
+
+# This file will also catch any and all errors from run_by_huc_wb, even script aborts from that file
+
+# You really can not call directly to run_unit_wb.sh as that file relys on export values
+# from this file.
+# run_by_huc_wb will futher process branches with its own iterator (parallelization).
+
+# Sample Usage: /foss_fim/fim_process_unit_wb.sh rob_test_wb_1 05030104
+
+## START MESSAGE ##
+
+echo
+
+usage ()
+{
+    echo
+    echo 'Produce FIM hydrofabric datasets for a single unit and branch scale.'
+    echo 'NOTE: fim_pre_processing must have been already run and this tool'
+    echo '      will not include post processing. Only single independent single'
+    echo '      huc and its branches.'    
+    echo 'Usage : There are no arg keys (aka.. no dashes)'
+    echo '        you need the run name first, then the huc.'
+    echo '        ie ) /foss_fim/fim_process_unit_wb.sh rob_test_1 05030104'
+    echo
+    exit
+}
+
+export runName=$1
+export hucNumber=$2
+
+# print usage if arguments empty
+if [ "$runName" = "" ]
+then
+    echo "ERROR: Missing run time name argument (1st argument)"
+    usage
+fi
+
+if [ "$hucNumber" = "" ]
+then
+    echo "ERROR: Missing hucNumber argument (2nd argument)"
+    usage
+fi
+
+re='^[0-9]+$'
+if ! [[ $hucNumber =~ $re ]] ; then
+   echo "Error: hucNumber is not a number" >&2; exit 1
+   usage
+fi
+
+echo "=========================================================================="
+echo "---- Start of huc processing for $hucNumber"
+
+# outputDataDir, srcDir and others come from the Dockerfile
+
+export outputRunDataDir=/$outputDataDir/$runName
+export outputHucDataDir=$outputRunDataDir/$hucNumber
+export outputBranchDataDir=$outputHucDataDir/branches
+export current_branch_id=0
+
+## huc data
+if [ -d "$outputHucDataDir" ]; then
+    rm -rf $outputHucDataDir
+fi
+
+# make outputs directory
+mkdir -p $outputHucDataDir
+mkdir -p $outputBranchDataDir
+
+# Clean out previous unit logs and branch logs starting with this huc
+rm -f $outputRunDataDir/logs/unit/"$hucNumber"_unit.log
+rm -f $outputRunDataDir/logs/branch/"$hucNumber"_summary_branch.log
+rm -f $outputRunDataDir/logs/branch/"$hucNumber"*.log
+rm -f $outputRunDataDir/unit_errors/"$hucNumber"*.log
+rm -f $outputRunDataDir/branch_errors/"$hucNumber"*.log
+hucLogFileName=$outputRunDataDir/logs/unit/"$hucNumber"_unit.log
+
+# Process the actual huc
+/usr/bin/time -v $srcDir/run_unit_wb.sh | tee $hucLogFileName
```

Comment:
When submitting a known bad huc to this script directly, the stderr/stdout terminal displays at what point and why the process stopped, however, the stderr/stdout is not making it to the `$hucLogFileName` log. This poses a problem, as when running `fim_pipeline.sh` with a bad huc, the error/exit code are not captured and a user might be under the impression the run was successful. 

`/usr/bin/time -v $srcDir/run_unit_wb.sh | tee $hucLogFileName`
could be replaced with :
`/usr/bin/time -v $srcDir/run_unit_wb.sh 2>&1 | tee $hucLogFileName`
to resolve this. 

See `/foss_fim/outputs/test_robhpr806_fimpreprocessing_ud/logs/unit/04300109_unit.log` for the original run. 
See `/foss_fim/outputs/test_robhpr806_fimpreprocessing_zd/logs/unit/04300109_unit.log` for a run with modification listed above.

Reply:
Great idea. Adding it now. Coming on next commit